src/assignmentNotification.test.js

```diff
@@ -112,46 +112,47 @@ describe('AssignmentNotification', () => {
 
       testContext.notification.send();
 
-      expect($.ajax).toHaveBeenCalledTimes(2);
-      expect($.ajax).toHaveBeenNthCalledWith(1, 'http://testtrack.dev/api/v1/assignment_event', {
-        method: 'POST',
-        dataType: 'json',
-        crossDomain: true,
-        data: {
+      expect(client.post).toHaveBeenCalledTimes(2);
+      expect(client.post).toHaveBeenNthCalledWith(
+        1,
+        'http://testtrack.dev/api/v1/assignment_event',
+        {
           visitor_id: 'visitorId',
           split_name: 'jabba',
           context: 'spec',
           mixpanel_result: undefined
-        }
-      });
-      expect($.ajax).toHaveBeenNthCalledWith(2, 'http://testtrack.dev/api/v1/assignment_event', {
-        method: 'POST',
-        dataType: 'json',
-        crossDomain: true,
-        data: {
+        },
+        { crossDomain: true }
+      );
+      expect(client.post).toHaveBeenNthCalledWith(
+        2,
+        'http://testtrack.dev/api/v1/assignment_event',
+        {
           visitor_id: 'visitorId',
           split_name: 'jabba',
           context: 'spec',
           mixpanel_result: 'failure'
-        }
-      });
+        },
+        { crossDomain: true }
+      );
     });
 
     it('logs an error if the request fails', () => {
-      $.ajax = jest.fn().mockImplementation(function() {
-        return $.Deferred().rejectWith(null, [
-          { status: 500, responseText: 'Internal Server Error' },
-          'textStatus',
-          'errorThrown'
-        ]);
+      client.post = jest.fn().mockRejectedValue({
+        response: {
+          status: 500,
+          statusText: 'Internal Server Error',
+          data: null
+        }
       });
 
-      testContext.notification.send();
-
-      expect(testContext.visitor.logError).toHaveBeenCalledTimes(1);
-      expect(testContext.visitor.logError).toHaveBeenCalledWith(
-        'test_track persistAssignment error: [object Object], 500, Internal Server Error, textStatus, errorThrown'
-      );
+      // eslint-disable-next-line jest/valid-expect-in-promise
+      testContext.notification.send().then(() => {
```

Comment:
We should return the Promise to tell `jest` there are pending expectations. This should also help get ride of the eslint warnings.
```suggestion
      return testContext.notification.send().then(() => {
```

Reply:
Having some trouble with this one. When I refactor it to return a promise, the test fails with a jest timeout. Haven't been able to figure out why just yet.

Reply:
I believe `send` is never resolving because the `testContext.analyticsTrackStub.mockImplementation` is never called for this test block so it times out waiting for a response. 

```js
    it('logs an error if the request fails', () => {
      testContext.analyticsTrackStub.mockImplementation((visitor_id, assignment, callback) => {
        callback(false);
      });
      client.post = jest.fn().mockRejectedValue({
        response: {
          status: 500,
          statusText: 'Internal Server Error',
          data: null
        }
      });

      return testContext.notification.send().then(() => {
        expect(testContext.visitor.logError).toHaveBeenCalledTimes(2);
        expect(testContext.visitor.logError).toHaveBeenCalledWith(
          'test_track persistAssignment error: 500, Internal Server Error, null'
        );
      });
    });
```